### ISSUE TITLE
Use pure JavaScript resize plugin

### ISSUE THREAD

**Issue by TheBosZ** [ManifestWebDesign/angular-gridster#145]
We currently have a dependency on a resize-event plugin. We're using the jQuery version but there is a pure JavaScript version available.

Need to redo the event binding based on the docs: https://github.com/sdecima/javascript-detect-element-resize

**Comment by viniciuspires** [comment]
Hi. I know it's not the same thing, but since this bug is an enhancement, can it be possible to add the functionality of manually triggering the gridster `resize` function?

The problem that I have is that the window is not resized itself, but instead the container of the gridster directive is resized, and it's not responding to that.

I need this asap, so I forked your repo to add this funcionality. If you guys think it's a valuable enhancement, I'd be honored to send the PR back :)

I'm thinking of adding this through a service, that sends a "resize" event that the gridster controller can listen. Does it look reasonable, or should I try another way?

Thanks in advance.

**Comment by mtraynham** [comment]
@viniciuspires Actually, this bug could be closed now based on the [recent changes to the source](https://github.com/ManifestWebDesign/angular-gridster/blob/master/src/angular-gridster.js#L775).  It should be irrelevant which resize script from `javascript-detect-element-resize` is being used.

All that being said, are you finding cases where an **element** resize is not triggering the gridster resize?  If so, it may be a [javascript-detect-element-resize](https://github.com/sdecima/javascript-detect-element-resize) bug instead.

**Comment by viniciuspires** [comment]
Oh, I see... I think the bug is that I'm not resizing the `gridster` element itself. I'm resizing the parent element which is a `ng-view`, like this:

```html
<div ng-view>
  <div gridster> ... </div>
</div>
```

I just have to know whether this is a bug with `javascript-detect-element-resize` or if it isn't a spec at all...

Anyway, I hope this helps anyone else that has this problem.

**Comment by viniciuspires** [comment]
Nope, sorry. There were several problems. First I forgot to include `javascript-detect-element-resize` in requirejs, and now it works fine with parent resizing AND css animation.

But it's still a little rough the way that it's done. It waits until the end of the animation to trigger the resize. I may be wrong, but I think angular-gridster would work like a charm using the host GPU if it worked with percentages when it comes to the width of the elements. This could be an opt-in feature...

What are your thoughts on that? Should I open a enhancement bug for that?

**Comment by paddyfay** [comment]
Hi. Is jQuery still a dependency for resizing?

**Comment by linxux** [comment]
@paddyfay Yes, it is jQuery-free.
Use pure javascript `Element Resize Detection` library instead of jQuery plugin,
> <script src="bower_components/javascript-detect-element-resize/detect-element-resize.js"></script>
